imx8mm_var-dart: fit_config: support ap20.dtb

lidaide opened the pull request:
ap20.dtb is a copy of ap20-pt1.5.dtb, it is a formal release name.
If board_id < D revision, u-boot will show "not support" and not boot again, others will boot from ap-20.dtb.

lidaide committed:
imx8mm_var-dart: fit_config: support ap20.dtb

ap20.dtb is a copy for ap20-pt1.5, it is a formal release name.
If board_id revision below D, it will not support again,
others will load ap20.dtb instead.

Signed-off-by: Daide Li <dai-de.li@leica-geosystems.com.cn>

grembeter commented on include/configs/imx8mm_var_dart.h:
```diff
@@ -89,8 +89,10 @@
 		"if test \"${board_id}\" < \"D\"; then" \
 		"    env set fit_config conf-freescale_ap20-pt1.dtb;" \
-		"else" \
+		"elif test \"${board_id}\" = \"D\"; then" \
 		"    env set fit_config conf-freescale_ap20-pt1-5.dtb;" \
+		"else" \
+		"    env set fit_config conf-freescale_ap20.dtb;" \
```
This assumption looks very strong. Please provide more info in commit message: is there really a difference between `ap20` and `ap20-pt1.5`?

lidaide replied:
keep pt1 and pt1-5. create a new ap20, which is a copy pt1-5
load pt1 for revision below D, load pt1-5 for revision D, load ap20 for revision above D.

renecharles replied:
@lidaide indeed, we have a problem here, i didn't think that through enough, sorry
this conflicts with removing the dtb files from the geosurv....

My recommendation would be to fully drop support for pt1 and pt1.5:

```
"if test \"${board_id}\" < \"D\"; then" \
		"    DISPLAY ERROR MESSAGE AND NO BOOT" \
		"else" \
		"    env set fit_config conf-freescale_ap20.dtb;" \
```
What do you think? @grembeter

grembeter reviewed (changes requested):
please remove `Jira: GDH-1500` tag from all commit messages. They are useless in public repositories.

lidaide commented:
V2:
1. update the comment message.

grembeter commented:
looks reasonable to me


g


________________________________
From: René-Charles ***@***.***>
Sent: Thursday, August 18, 2022 1:07:38 PM
To: hexagon-geo-surv/u-boot-imx ***@***.***>
Cc: Grygorii Tertychnyi ***@***.***>; Mention ***@***.***>
Subject: Re: [hexagon-geo-surv/u-boot-imx] imx8mm_var-dart: fit_config: support ap20.dtb (PR #19)


@renecharles commented on this pull request.

________________________________

In include/configs/imx8mm_var_dart.h<https://github.com/hexagon-geo-surv/u-boot-imx/pull/19#discussion_r948995963>:

> -             "else" \
+               "elif test \"${board_id}\" = \"D\"; then" \
                "    env set fit_config conf-freescale_ap20-pt1-5.dtb;" \
+               "else" \
+               "    env set fit_config conf-freescale_ap20.dtb;" \


@lidaide<https://github.com/lidaide> indeed, we have a problem here, i didn't think that through enough, sorry
this conflicts with removing the dtb files from the geosurv....

My recommendation would be to fully drop support for pt1 and pt1.5:

"if test \"${board_id}\" < \"D\"; then" \
                "    DISPLAY ERROR MESSAGE AND NO BOOT" \
                "else" \
                "    env set fit_config conf-freescale_ap20.dtb;" \


What do you think? @grembeter<https://github.com/grembeter>

—
Reply to this email directly, view it on GitHub<https://github.com/hexagon-geo-surv/u-boot-imx/pull/19#discussion_r948995963>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJXKJNB2MUKPIDATF7IEKY3VZYKPVANCNFSM56KTZR6Q>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

zandrey commented on include/configs/imx8mm_var_dart.h:
```diff
@@ -88,9 +88,9 @@
 	"get_fit_config=leica_mcu_comm board_id board_id;" \
 		"if test \"${board_id}\" < \"D\"; then" \
-		"    env set fit_config conf-freescale_ap20-pt1.dtb;" \
+		"    DISPLAY ERROR MESSAGE AND NO BOOT" \
```
NAK: This is neither a _U-Boot_ command nor meaningful message. Since this gets executed - it **will** lead to an error, so it has to be re-worked.

lidaide replied:
it is my fault, i have updated to `echo !!! No support board_id:\"{board_id}\", boot failed;`, and i have test on my evk board, it can show msg in the serial port. thank you.

lidaide commented:
V3:
update: `echo !!! No support board_id:\"{board_id}\", boot failed;`
when the board_id revision <D, it will show up when boot.

zandrey commented on include/configs/imx8mm_var_dart.h:
```diff
@@ -88,9 +88,9 @@
 	"get_fit_config=leica_mcu_comm board_id board_id;" \
 		"if test \"${board_id}\" < \"D\"; then" \
-		"    env set fit_config conf-freescale_ap20-pt1.dtb;" \
+		"    echo !!! No support board_id:\"${board_id}\", boot failed;" \
```
Please re-phrase to:
```shell
echo !!! ERROR: Board with ID ${board_id} is not supported;
```

This is more informative for the user.

lidaide committed:
imx8mm_var_dart: support ap20 downgrade ap20-pt1-5

ap20 is formal release name, ap20-pt1-5 is previous version.
If dragonstaff bump up, fit config will use ap20.dtb.
If dragrastaff bump down, fit config will use ap20-pt1-5.dtb.

when bump down, trigger_swupdate will split into 2 steps,
first step will upgrade image-updater, it only contains ap20-pt1-5.dtb,
but fit_config will point to ap20.dtb.
after set "swu=1", system will reboot to enter step2.

in the step2, the image-updater(old) will boot up, it will call
`bootm ${initrd_addr}#${fit_config}`, so fit_config=ap20-pt1-5.dtb,
system will boot up and continue.

Signed-off-by: Daide Li <dai-de.li@leica-geosystems.com.cn>

lidaide commented:
V4:
   support downgrade from `ap20` to `ap20-pt1.5`

zandrey commented on include/configs/imx8mm_var_dart.h:
```diff
@@ -87,10 +87,12 @@
+		"if test \"${board_id}\" < \"D\"; then " \
+		    "echo !!! ERROR: Board with ID ${board_id} is not supported; " \
+		"elif test -n \"${downgrade_dtb}\"; then " \
+		    "env set fit_config conf-freescale_${downgrade_dtb}; " \
```
I cannot seems to find `${downgrade_dtb}` variable. Where is it defined?

lidaide replied:
HI @zandrey , the `${downgrade_dtb}` variable was defined in the geo_surv, please see the PR `https://github.hexagon.com/geo-surv/meta-geosurv/pull/332/files`: 
`        if [ "$hw" = "AP20-PT1.5" ] && [ "$hw_model" = "AP20" ];
        then
            hw_compatible="Y"
            bootenv_set downgrade_dtb "ap20-pt1-5.dtb"
            break
        fi
`
